### ISSUE TITLE
Provide API for extensions to move cursor without affecting history

### ISSUE THREAD

**Issue by mltony** [microsoft/vscode#106603]
Hello!
I would like to be able to write an extension, that can move the cursor without affecting cursor history. To the best of my knowledge the only way to move the cursor from within extension is:
```
editor.selection = new vscode.Selection(...);
```
However, this affects the history of cursor navigation, as it would add an entry to cursor history list, which will affect where Alt+Left command will take you. Here is hwy I need it: I am a blind VSCode user; in order to navigate around source code files more effectively, I wrote IndentNav extension (https://github.com/mltony/vscode-indent-nav/). Alt+Up/Down takes you to next/previous line with the same indentation level. However, this also pollutes cursor navigation history, so that while I'm using IndentNav, Alt+Left arrow never takes you to the previous file, but rather, to the previous location within the same file where I have happened to be using IndentNav. Thus, I can either use IndentNav, or be able to navigate back and forth using Alt+Left/Right, but not both at the same time. IndentNav commands should be treated more like cursor navigation commands, e.g. next/previous line, but I found no way of achieving this from within extension.

I looked at the source code of move by word commands of VSCode, and I saw that they update the cursor via:
```
editor._getCursors().setStates('moveByWordCommand', CursorChangeReason.NotSet, result.map(r => CursorState.fromModelSelection(r))); 
```
However this way doesn't seem to work from within extension, as this API is not exposed.
I also feel that this issue is accessibility-related, since I have an NVDA add-on IndentNav, but it doesn't work with VSCode, since VSCode only provides a few lines of text at a time in accessibility mode, so cc @isidorn. If this accessibility restriction could be fixed - I won't have a need for a separate extension and this feature request would no longer be relevant.

**Comment by isidorn** [comment]
@mltony thanks for filling this. I think this request makes sense for your use case, however for exposing this via API @alexdima would be better to comment. Idealy the `Selection` could have some flag such that the extension could control if it should be stored in history. Though not sure if there are other requests for this.

As for a few lines of text at a time. I suggest that you tune the following setting `"editor.accessibilityPageSize"` and increase it to a 1000. We have this limiation due to https://github.com/microsoft/vscode/issues/90976

**Comment by alexdima** [comment]
I don't think this could be tackled by changing the `vscode.Selection` object, but perhaps by setting the selection in some different way (perhaps with a new method on the `TextEditor`).

**Comment by isidorn** [comment]
@mltony yes, that is due to the chrome limitation which is linked in issue #90976
